backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/ServerInfoSynchronizer.java

```diff
@@ -19,28 +19,36 @@
  */
 package org.sonarsource.sonarlint.core.serverconnection;
 
+import org.sonarsource.sonarlint.core.commons.Version;
 import org.sonarsource.sonarlint.core.commons.progress.SonarLintCancelMonitor;
 import org.sonarsource.sonarlint.core.serverapi.ServerApi;
 
 public class ServerInfoSynchronizer {
+  private static final String MQR_MODE_SETTING = "sonar.multi-quality-mode.enabled";
+  private static final String MQR_MODE_SETTING_MIN_VERSION = "10.8";
   private final ConnectionStorage storage;
 
   public ServerInfoSynchronizer(ConnectionStorage storage) {
     this.storage = storage;
   }
 
   public StoredServerInfo readOrSynchronizeServerInfo(ServerApi serverApi, SonarLintCancelMonitor cancelMonitor) {
-    return storage.serverInfo().read()
+    return storage.serverInfo().read(serverApi.isSonarCloud())
       .orElseGet(() -> {
         synchronize(serverApi, cancelMonitor);
-        return storage.serverInfo().read().get();
+        return storage.serverInfo().read(serverApi.isSonarCloud()).get();
       });
   }
 
   public void synchronize(ServerApi serverApi, SonarLintCancelMonitor cancelMonitor) {
     var serverStatus = serverApi.system().getStatus(cancelMonitor);
     var serverVersionAndStatusChecker = new ServerVersionAndStatusChecker(serverApi);
     serverVersionAndStatusChecker.checkVersionAndStatus(cancelMonitor);
-    storage.serverInfo().store(serverStatus);
+    var version = Version.create(serverStatus.getVersion());
+    String isMQRMode = null;
+    if (!serverApi.isSonarCloud() && version.compareToIgnoreQualifier(Version.create(MQR_MODE_SETTING_MIN_VERSION)) >= 0) {
```

Comment:
It's to avoid calling this endpoint when we know it does not exist, otherwise 404 error is thrown, WDYT?